addons/website_slides/models/slide_channel.py

```diff
@@ -35,29 +35,32 @@ def _compute_completion(self):
             mapped_data.setdefault(item['channel_id'][0], dict())
             mapped_data[item['channel_id'][0]][item['partner_id'][0]] = item['__count']
 
-        channel_data = {}
-        channel_ids = mapped_data.keys()
-        if channel_ids:
-            channel_read_res = self.env['slide.channel'].sudo().browse(channel_ids).read(['total_slides'])
-            channel_data = dict((channel['id'], channel['total_slides']) for channel in channel_read_res)
-
         for record in self:
             slide_done = mapped_data.get(record.channel_id.id, dict()).get(record.partner_id.id, 0)
-            slide_total = channel_data.get(record.channel_id.id) or 1
-            record.completion = math.ceil(100.0 * slide_done / slide_total)
+            record.completion = slide_done
 
     def _write(self, values):
         partner_karma = False
-        if 'completion' in values and values['completion'] >= 100:
-            values['completed'] = True
-            incomplete_channel_partners = self.filtered(lambda cp: not cp.completed)
-            partner_karma = dict.fromkeys(incomplete_channel_partners.mapped('partner_id').ids, 0)
-            for channel_partner in incomplete_channel_partners:
+        to_complete = self.env['slide.channel.partner']
+        if values.get('completion'):
+            incomplete_self = self.filtered(lambda cp: not cp.completed)
+            channels_data = {result['id']: result['total_slides'] for result in incomplete_self.mapped('channel_id').read(['total_slides'])}
+            for cp in incomplete_self:
+                if values.get('completion') >= channels_data[cp.channel_id.id]:
+                    to_complete |= cp
+
+            partner_karma = dict.fromkeys(incomplete_self.mapped('partner_id').ids, 0)
+            for channel_partner in incomplete_self:
```

Comment:
If I'm not mistaken you add karma for all incomplete channels, which doesn't seem right.

Reply:
It seems you were right. I added some tests and fixed the issue.